### PR TITLE
Add `clampToGround` to kml

### LINKED ISSUE #6799: Losing Z values when multiple KML files are overlaid
When I drag and drop the same KML file twice, then it looks like the second data lose the Z value on the map. 
In the following image and video, those two triangles are the exactly the same KML data. I added #1 first followd by #2.
I attached the KML file I tested.

image
[test.zip](https://github.com/TerriaJS/terriajs/files/11673243/test.zip)

https://github.com/TerriaJS/terriajs/assets/4824436/a06a7b17-02e8-4192-8ff8-1be069863504

### PR DESCRIPTION
**Pull request by nf-s**
### Add `clampToGround` to kml

Fixes https://github.com/TerriaJS/terriajs/issues/6799

Related to 

- https://github.com/TerriaJS/terriajs/issues/4958

Terria used to clamp 2D and 3D KML polygon features to ground, and there was no way to turn it off. I have removed Terria's clamping, added a `clampToGround` trait, and now we pass that into `KmlDataSource.load()` - where cesium will now handle clamping

`clampToGround` will default to `true` - but Cesium will respect `altitudeMode` in KML - so it won't clamp 3D polygons that have `altitudeMode` set to `absolute` or `relativeToGround`.

There is a weird behaviour where KML polygons will only be clamped if in 3d (cesium) mode, this is because when `loadMapItems` is loaded, cesium terrain is sampled to clamp polygon features. This is now fixed.

### Test me

```json
{
  "name": "3D KML Test clampToGround = false",
  "type": "kml",
  "url": "https://gist.githubusercontent.com/nf-s/cfa170f067c8c462467558e9971f811d/raw/acf861a79fd206770029bb739fb76b5b84080e98/test-polygon.kml",
  "clampToGround": false
},
{
  "name": "3D KML Test clampToGround = true",
  "type": "kml",
  "url": "https://gist.githubusercontent.com/nf-s/cfa170f067c8c462467558e9971f811d/raw/acf861a79fd206770029bb739fb76b5b84080e98/test-polygon.kml",
  "clampToGround": true
},
{
  "name": "2D KML Test clampToGround = false",
  "type": "kml",
  "url": "https://gist.githubusercontent.com/nf-s/e0174e6f4114b9bab01307fe183be1e2/raw/7c2e5c210d8b7f0120d961d4e6086ad92b0caf44/test-2d-polygon.kml",
  "clampToGround": false
},
{
  "name": "2D KML Test clampToGround = true",
  "type": "kml",
  "url": "https://gist.githubusercontent.com/nf-s/e0174e6f4114b9bab01307fe183be1e2/raw/7c2e5c210d8b7f0120d961d4e6086ad92b0caf44/test-2d-polygon.kml",
  "clampToGround": true
}
```

- [Before link](http://ci.terria.io/main/#clean&start=%7B%22initSources%22%3A%5B%7B%22catalog%22%3A%20%5B%7B%22name%22%3A%20%223D%20KML%20Test%22%2C%22type%22%3A%20%22kml%22%2C%22url%22%3A%20%22https%3A%2F%2Fgist.githubusercontent.com%2Fnf-s%2Fcfa170f067c8c462467558e9971f811d%2Fraw%2Facf861a79fd206770029bb739fb76b5b84080e98%2Ftest-polygon.kml%22%7D%2C%7B%22name%22%3A%20%222D%20KML%20Test%22%2C%22type%22%3A%20%22kml%22%2C%22url%22%3A%20%22https%3A%2F%2Fgist.githubusercontent.com%2Fnf-s%2Fe0174e6f4114b9bab01307fe183be1e2%2Fraw%2F7c2e5c210d8b7f0120d961d4e6086ad92b0caf44%2Ftest-2d-polygon.kml%22%7D%5D%7D%5D%7D)
  - Notice that all polygons are clamped to the ground
- [After link](http://ci.terria.io/kml-clamp-polyhon/#clean&start=%7B%22initSources%22%3A%5B%7B%22catalog%22%3A%20%5B%7B%22name%22%3A%20%223D%20KML%20Test%20clampToGround%20%3D%20false%22%2C%22type%22%3A%20%22kml%22%2C%22url%22%3A%20%22https%3A%2F%2Fgist.githubusercontent.com%2Fnf-s%2Fcfa170f067c8c462467558e9971f811d%2Fraw%2Facf861a79fd206770029bb739fb76b5b84080e98%2Ftest-polygon.kml%22%2C%22clampToGround%22%3A%20false%7D%2C%7B%22name%22%3A%20%223D%20KML%20Test%20clampToGround%20%3D%20true%22%2C%22type%22%3A%20%22kml%22%2C%22url%22%3A%20%22https%3A%2F%2Fgist.githubusercontent.com%2Fnf-s%2Fcfa170f067c8c462467558e9971f811d%2Fraw%2Facf861a79fd206770029bb739fb76b5b84080e98%2Ftest-polygon.kml%22%2C%22clampToGround%22%3A%20true%7D%2C%7B%22name%22%3A%20%222D%20KML%20Test%20clampToGround%20%3D%20false%22%2C%22type%22%3A%20%22kml%22%2C%22url%22%3A%20%22https%3A%2F%2Fgist.githubusercontent.com%2Fnf-s%2Fe0174e6f4114b9bab01307fe183be1e2%2Fraw%2F7c2e5c210d8b7f0120d961d4e6086ad92b0caf44%2Ftest-2d-polygon.kml%22%2C%22clampToGround%22%3A%20false%7D%2C%7B%22name%22%3A%20%222D%20KML%20Test%20clampToGround%20%3D%20true%22%2C%22type%22%3A%20%22kml%22%2C%22url%22%3A%20%22https%3A%2F%2Fgist.githubusercontent.com%2Fnf-s%2Fe0174e6f4114b9bab01307fe183be1e2%2Fraw%2F7c2e5c210d8b7f0120d961d4e6086ad92b0caf44%2Ftest-2d-polygon.kml%22%2C%22clampToGround%22%3A%20true%7D%5D%7D%5D%7D)
  - Notice that all 3D polygons are not clamped to the ground (regardless of `clampToGround` value)
  - Notice that 2D polygons are clamped when `clampToGround` is true (set mode to 3d terrain)

### Checklist

- [ ] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] I've updated relevant documentation in `doc/`.
- [x] I've updated CHANGES.md with what I changed.
- [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

**Comment by imakihi** [comment]
@nf-s 
"clampPolygonsToGround": false worked!!!
Thank you for making this work.
Best.
Hiroo

**Comment by imakihi** [comment]
@nf-s 
Is it possible to set clampPolygonsToGround = false as a default? 
I wanted to locally import KML files and overlay those in the 3D mode (not being clumped).
What do you think?

Hiroo

**Comment by nf-s** [comment]
Hi @imakihi

I looked into this a bit more and I have removed the polygon clamp feature. Cesium now has a `clampToGround` flag that applies its own logic to KML files.

This means that KML files with 3D polygons that have `altitudeMode` set to `absolute` or `relativeToGround` will never be clamped - regardless of `clampToGround`.

We still need to do some testing - but I think this is a better way forward than before, where Terria was manually overriding coordinate heights.

What do you think @steve9164 ?

**Comment by imakihi** [comment]
Hi @nf-s 
Thank you for looking into this. Your logic makes sense to me.
Please let me know if I can test this new implementation.

**Comment by nf-s** [comment]
> Hi @nf-s Thank you for looking into this. Your logic makes sense to me. Please let me know if I can test this new implementation.

Hi @imakihi 

That would be great, thanks!
If you have a few different sources of KML, can you please drag them into the test deployment (http://ci.terria.io/kml-clamp-polyhon) and let us know if you see any weird behavior with the new `clampToGround` (this will be set to `true` by default)?

**Comment by imakihi** [comment]
@nf-s 
I tested 5 kml files with different altitude modes.
- absolute
- clump to ground
- clump to seafloor
- relative to ground
- relative to seafloor

Except relative to ground, I think, it worked. When I drag and drop my test file for the relative to ground, it doesn't clump, I think.
I created my sample KML file from here. 
https://developers.google.com/kml/documentation/altitudemode

These are my test files.
[altitudemode.zip](https://github.com/TerriaJS/terriajs/files/13331456/altitudemode.zip)

**Review by tephenavies — COMMENTED** [review]
The code changes look good. Have you had time to test the KMLs?

**Comment by nf-s** [comment]
I tested the KMLs, but couldn't find a difference between `clampToGround` = `true` or `false`

To be honest, I find it quite difficult to figure out exactly how `clampToGround` works in cesium - the code path is pretty annoying to follow.

Basically - if `clampToGround` is true

- [billboards/labels will get a `heightReference` based on altitude mode](https://github.com/CesiumGS/cesium/blob/3d8f53819fcf942f2bfbebaedfa65cae45365e2a/packages/engine/Source/DataSources/KmlDataSource.js#L1534-L1538)
- [`PolylineGraphics` `clampToGround` will be set to true (if not extruded)](https://github.com/CesiumGS/cesium/blob/3d8f53819fcf942f2bfbebaedfa65cae45365e2a/packages/engine/Source/DataSources/KmlDataSource.js#L1654-L1669)
- [`PolygonGraphics` will get `height = 0` (if not extruded)](https://github.com/CesiumGS/cesium/blob/3d8f53819fcf942f2bfbebaedfa65cae45365e2a/packages/engine/Source/DataSources/KmlDataSource.js#L1752-L1754)

Anyway, I think we should just merge this in. This respects KML features (and altitude mode), and correctly passes `clampToGround` into cesium.

**Comment by imakihi** [comment]
@nf-s 
I have one LineString KML file and tested clampToGround=true but it didn't clamp. I didn't see any issues on polygons... 
Can you think of any reasons?

I used following two KMLs for my test.
https://s3.ap-northeast-1.amazonaws.com/3dimension.jp/tmp/imaki_test/layer_clamp_line.kml
https://s3.ap-northeast-1.amazonaws.com/3dimension.jp/tmp/imaki_test/layer_clamp_polygon.kml

This is my catalog.json
[kml_test.json](https://github.com/user-attachments/files/16949302/kml_test.json)

Thanks. Hiroo

File: CHANGES.md [hunk 1]
```diff
@@ -51,6 +53,7 @@
 - Fix bug in mismatched GeoJSON Feature `_id_` and TableMixin `rowId` - this was causing incorrect styling when using `filterByProperties` or features had `null` geometry
 - Fix splitter for `GeoJsonMixin` (lines and polygon features only)
 - Fix share links with picked features from `ProtomapsImageryProvider`
+- Add `hideDefaultDescription` to `CatalogMemberTraits` - if true, then no generic default description will be shown when `description` is empty.
```
Comment (ljowen) [inline CHANGES.md:230]:
```suggestion
```

**Review by ljowen — APPROVED** [review]

**Pull request merged**